developer-extension/src/content-scripts/main.ts

```diff
@@ -1,11 +1,128 @@
-// This script is executed in the "main" execution world, the same world as the webpage. Thus, it
-// can define a global callback variable to listen to SDK events.
-
-;(window as any).__ddBrowserSdkExtensionCallback = (message: unknown) => {
-  // Relays any message to the "isolated" content-script via a custom event.
-  window.dispatchEvent(
-    new CustomEvent('__ddBrowserSdkMessage', {
-      detail: message,
-    })
-  )
+import { EventListeners } from '../common/eventListeners'
+import { DEV_LOGS_URL, DEV_RUM_URL, SESSION_STORAGE_SETTINGS_KEY } from '../common/constants'
+
+declare global {
+  interface Window extends EventTarget {
+    DD_RUM?: SdkPublicApi
+    DD_LOGS?: SdkPublicApi
+    __ddBrowserSdkExtensionCallback?: (message: unknown) => void
+  }
+}
+
+type SdkPublicApi = { [key: string]: (...args: any[]) => unknown }
+
+function main() {
+  if (window.__ddBrowserSdkExtensionCallback) {
+    return
+  }
+
+  sendEventsToExtension()
+
+  const stringSettings = sessionStorage.getItem(SESSION_STORAGE_SETTINGS_KEY)
+  const settings = stringSettings && JSON.parse(stringSettings)
```

Comment:
<!-- Suggestions are specific requests to improve the subject under review. It is assumed that we all want to do what's best, so these comments are never dismissed as “mere suggestions”, but are taken seriously.  -->
**💬 suggestion:** what about try/catching the JSON.parse?‏